Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -240,39 +240,17 @@ def audit_style
     end
 
     def audit_file
-      actual_mode = formula.path.stat.mode
-      # Check that the file is world-readable.
-      if actual_mode & 0444 != 0444
-        problem format("Incorrect file permissions (%03<actual>o): chmod %<wanted>s %<path>s",
-                       actual: actual_mode & 0777,
-                       wanted: "+r",
-                       path:   formula.path)
-      end
-      # Check that the file is user-writeable.
-      if actual_mode & 0200 != 0200
-        problem format("Incorrect file permissions (%03<actual>o): chmod %<wanted>s %<path>s",
-                       actual: actual_mode & 0777,
-                       wanted: "u+w",
-                       path:   formula.path)
-      end
-      # Check that the file is *not* other-writeable.
-      if actual_mode & 0002 == 002
-        problem format("Incorrect file permissions (%03<actual>o): chmod %<wanted>s %<path>s",
-                       actual: actual_mode & 0777,
-                       wanted: "o-w",
-                       path:   formula.path)
-      end
-
+      # TODO: check could be in RuboCop
       problem "'DATA' was found, but no '__END__'" if text.data? && !text.end?
```

Comment:
I wanted to give the those checks a try but I'm not sure how one would access the necessary `text` contents in `Rubocop`. Can you give me hint please @MikeMcQuaid ?
And would it belong in `patches.rb` or `files.rb`?

Reply:
I'd say probably `patches.rb`. If you check out the definitions in https://github.com/Homebrew/brew/blob/2e801c3077e362562b0e2717a5d219a474958ca3/Library/Homebrew/dev-cmd/audit.rb#L167-L173 that may help. I'm not 100% how to get the source of the entire file in `rubocop`; there may also be a way to only get the `__END__` section from the file from `rubocop` (or `parser.rb` that it uses).

Reply:
With my limited knowledge about the codebase and tools I haven't figured it out for now.
I don't see a straight forward solution to read the `__END__` section from the things available in `audit_formula` in `RuboCop`, i.e. the `body` only contains all the stuff that makes up the `class`, but not the stuff (i.e. `__END__`) after it :)

Reply:
Check out: https://www.rubydoc.info/gems/rubocop/0.46.0/RuboCop/ProcessedSource#lines-instance_method
https://www.honeybadger.io/blog/data-and-end-in-ruby/
https://github.com/rubocop-hq/rubocop/issues/1541

As one of them may help. It looks like you may need to look for usage of `patch :DATA`, get the filename and then do a full read of the file looking for `^__END__$`. Make sense?

Reply:
Late follow-up :) 
I'm fully back at work and didn't have a time to look at this again so far and probably won't have in the near future.

Reply:
No worries, someone else can pick it up!